aiida/restapi/common/utils.py

```diff
@@ -682,7 +682,7 @@ def parse_query_string(self, query_string):
         )
         # Value types
         value_num = ppc.number
-        value_bool = (Literal('true') | Literal('false')).addParseAction(lambda toks: bool(toks[0]))
+        value_bool = (Literal('true') | Literal('false')).addParseAction(lambda toks: toks[0] == 'true')
```

Comment:
```suggestion
        value_bool = (Literal('true') | Literal('false')).addParseAction(lambda toks: toks[0])
```
I think this should be enough. If you look at https://github.com/aiidateam/aiida-core/blob/cb82327e5e4587b8752674a6b57e64613a5e11f2/aiida/restapi/resources.py#L459, it will check for both `'false'` and `'False'` as a string.

As for test, you can check `tests/restapi/test_routes.py::TestRestApi::test_structure_download` as an example.

Reply:
Thanks @unkcpz. I think the idea of the line

https://github.com/aiidateam/aiida-core/blob/2c183fc4486e00f3348a1b66cdcd6d9fbfd563f0/aiida/restapi/common/utils.py#L685

is to convert the strings 'true' and 'false' in the query parameters into python boolean values True and False. This is what the previous code intended to do (`bool` conversion) but it unintentionally converted 'false' also to True.

If we adopt your change, you would keep these as strings. In the end it would work, as indeed the line in resources.py also checks for the string variants, but i think it's not the intended way.

Reply:
Yes, I agree with you. Then it can be `download not in ['false', 'False', False]` -> `download`. Just mention that these two parts are related.

Reply:
Well, the point is that the uppercase "False" and "True" in the query parameters are still interpreted as strings, so I would keep the line in resources.py as it was.

Reply:
Okay, I see. `Literal` of pyparsing will only catch `'true'` and `'false'` in line 685. It makes sense to me now. If you can add test from #5576 with `download=false`, I think I can approve this and merge it.

Reply:
Or probably simply merge this as it is and then test it in #5576 with `download=false`? This is why I was advising combining these two PRs.

Reply:
let's merge #5576 first, as this code won't without it. Regarding testing, i'm not sure what's the best way. The safest would be to add a separate test for lower and upper case versions (`download=False` and `download=false`).